orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -425,6 +575,83 @@ class TaskController {
     )
   }
 
+  @PostAuthorize("hasPermission(returnObject.application, 'APPLICATION', 'READ')")
+  private List<String> getPipelineConfigIdsOfReadableApplications() {
+    List<String> applicationNames = front50Service.getAllApplications()*.name as List<String>
+    List<String> pipelineConfigIds = applicationNames.stream()
+      .map{ applicationName -> front50Service.getPipelines(applicationName, false)*.id as List<String> }
+      .flatMap{ c -> c.stream() }
+      .collect(Collectors.toList())
+
+    return pipelineConfigIds
+  }
+
+  private static boolean recursivelyCheckIfObjectFieldsMatchesAllSubsetFields(Object object, Object subset) {
+    if (String.isInstance(object) && String.isInstance(subset)) {
+      // object matches subset if:
+      // - object equals subset
+      // - object matches subset as a regular expression
+      return ((String) object).equalsIgnoreCase((String) subset)
+    } else if (Map.isInstance(object) && Map.isInstance(subset)) {
+      // object matches subset if:
+      // - object contains all keys of subset and their values match
+      Map objectAsMap = (Map) object
+      Map subsetAsMap = (Map) subset
+      for (Object key : subsetAsMap.keySet()) {
+        if (!recursivelyCheckIfObjectFieldsMatchesAllSubsetFields(objectAsMap.get(key), subsetAsMap.get(key))) {
+          return false
+        }
+      }
+      return true
+    } else if (Collection.isInstance(object) && Collection.isInstance(subset)) {
+      // object matches subset if:
+      // - object contains a unique item that matches each item in subset
+      //   * this means that an item in subset may not match to more than one item in object, which
+      //     means that we should check every permutation of object to avoid greedily stopping
+      //     on the first item that matches.
+      //     e.g., Given,
+      //             object: [ { "name": "a", "version": "1" }, { "name" } ]
+      //           Without checking all permutations, this will match:
+      //             subset: [ { "name": "a", "version": "1" }, { "name" } ]
+      //           but will not match:
+      //             subset: [ { "name": "a" }, { "name", "version": "1" } ]
+      //           because the first item in subset will greedily match the first item in object,
+      //           leaving the second items in both, which do not match. This is fixed by checking
+      //           all permutations of object.
+      Collection<List> permutationsOfObjectAsList = Collections2.permutations((Collection) object)
```

Comment:
Generating all permutations in memory feels very expensive... Is there a way to generate permutations one by one and exit early?